src/app/views/fireside/_sidebar/AppFiresideSidebarFiresideSettings.vue

```diff
@@ -52,13 +55,57 @@ const {
 	canReport,
 } = c;
 
+const selectedCommunities = ref<{ community: Community }[]>([]);
+const targetableCommunities = computed(() => selectedCommunities.value.map(i => i.community));
+const selectedRealms = ref<Realm[]>([]);
+const maxRealms = ref(5);
+
 const form: FormController<Fireside> = createForm({
 	warnOnDiscard: false,
 	modelClass: Fireside,
 	// Just wrapping in a ref to make the form happy. It never actually changes.
 	model: ref(fireside),
+	loadUrl: `/web/dash/fireside/save/${fireside.hash}`,
+	onLoad: response => {
+		maxRealms.value = response.maxRealms;
+	},
+	onInit: () => {
+		selectedCommunities.value = fireside.community_links.map(i => {
+			return { community: i.community };
+		});
+		selectedRealms.value = fireside.realms.map(i => i.realm);
+	},
+	onSubmit: () => form.formModel.$saveWithRealms(selectedRealms.value.map(i => i.id)),
+	onSubmitSuccess: () => {
+		fireside.assign(form.formModel);
+	},
 });
 
+function attachRealm(realm: Realm, append = true) {
+	// Do nothing if that realm is already attached.
+	if (selectedRealms.value.find(i => i.id === realm.id)) {
+		return;
+	}
+
+	if (append) {
+		selectedRealms.value.push(realm);
+	} else {
+		selectedRealms.value.unshift(realm);
+	}
+	form.changed = true;
+}
+
+function removeRealm(realm: Realm) {
+	const idx = selectedRealms.value.findIndex(i => i.id === realm.id);
+	if (idx === -1) {
+		console.warn('Attempted to remove a realm that is not attached');
+		return;
+	}
+
+	selectedRealms.value.splice(idx, 1);
```

Comment:
Change to `arrayRemove`. Can check the results of the function to do the same thing if needed.